gprofiler/profilers/java.py

```diff
@@ -498,6 +501,8 @@ def _run_async_profiler(self, cmd: List[str]) -> None:
                 raise JattachSocketMissingException(*args) from None
             else:
                 raise JattachException(*args) from None
+        else:
+            logger.debug("async-profiler log", jattach_cmd=cmd, ap_log=self._read_ap_log())
```

Comment:
Why we only capture the AP log after `success`? 
And also, do we need the entire log each time? won't it be better to log the diff lines from before and after the profiling session? and filter the log lines by PID ?

Reply:
What do you mean "after `success`"?

The log is read in the `else` clause upon success, or in the `except` caluse before we raise an exception.

> And also, do we need the entire log each time? won't it be better to log the diff lines from before and after the profiling session? and filter the log lines by PID ?

The log is per-pid and is deleted after every async-profiler operation (so each log we read is the log for this single run)